backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/DriftEmailPayloadAggregator.java

```diff
@@ -76,4 +78,15 @@ public Integer getUniqueHostCount() {
         return this.uniqueHosts.size();
     }
 
+    public Boolean hasSystem(JsonObject drift, String baselineId, String insightsId) {
+        JsonObject baseline = drift.getJsonObject(baselineId);
+        JsonArray systems = baseline.getJsonArray(HOST_KEY);
+        for (int i = 0; i < systems.size(); i++) {
+            if (systems.getJsonObject(i).getString(INVENTORY_ID) == insightsId) {
```

Comment:
Should use equals to verify 
```suggestion
            if (systems.getJsonObject(i).getString(INVENTORY_ID).equals(insightsId)) {
```